pkg/controller/bastion/actuator_reconcile.go

```diff
@@ -461,3 +469,32 @@ func findNextFreeNumber(set map[int32]bool, baseValue int32) *int32 {
 	set[baseValue] = true
 	return &baseValue
 }
+
+func getLatestSku(ctx context.Context, opt *Options, factory azureclient.Factory) (string, error) {
+	vmImageclient, err := factory.VirtualMachineImage(ctx, opt.SecretReference)
+	if err != nil {
+		return "", err
+	}
+
+	result, err := vmImageclient.ListSkus(ctx, opt.Location, *vmImage("").Publisher, *vmImage("").Offer)
+	if err != nil {
+		return "", err
+	}
+
+	sku := sets.NewString()
+	re := regexp.MustCompile(`\d{2}\.\d{2}$`)
+	// regex only xx.xx version eg 18.04, 19.04
+	for _, v := range *result.Value {
+		if re.MatchString(*v.Name) {
+			sku.Insert(*v.Name)
+		}
+
+	}
+
+	if sku.List()[len(sku.List())-1] == "" {
+		return "", errors.New("sku not found")
+	}
+
+	return sku.List()[len(sku.List())-1], nil
```

Comment:
Maybe I am confused. Is the ordering - the "latest" as in the function name - implicit on the order of results returned by the API ? Because I see the filtering done here but not the ordering necessarily.

Reply:
yes it is  implicit on the order of results returned by the API as tested from my local for the `sku.List()` values
<img width="470" alt="image" src="https://github.com/gardener/gardener-extension-provider-azure/assets/42234376/6865ab09-4087-4627-9fd1-5b3d26fd2da9">